src/main/kotlin/com/cognifide/gradle/aem/tooling/tail/Tailer.kt

```diff
@@ -0,0 +1,44 @@
+package com.cognifide.gradle.aem.tooling.tail
+
+import java.io.BufferedReader
+
+class Tailer(private val source: LogSource, private val destination: LogDestination, val name: String = "") {
+
+    private val parser = Parser()
+
+    private var lastLogChecksum = ""
+
+    fun tail() {
+        val logs = source.processChunk { chunkReader ->
+            parser.parseLogs(chunkReader)
+        }
+        val newLogs = determineNewLogs(logs)
+
+        if (newLogs.isNotEmpty()) {
+            lastLogChecksum = newLogs.last().checksum
+            destination.dump(newLogs)
+        }
+    }
+
+    private fun determineNewLogs(logs: List<Log>): List<Log> {
+        val reversedLogs = logs.reversed()
```

Comment:
Is it memory optimal? How big this list will be?

Reply:
At the moment it is 400. I changed this to `logs.asReversed()` which is read only view, not consuming additiona memory.